core/devmode/src/main/java/io/quarkus/dev/RuntimeUpdatesProcessor.java

```diff
@@ -354,7 +353,7 @@ private String getFileExtension(File file) {
                         if (value > existing) {
                             ret.add(path);
                             log.infof("File change detected: %s", file);
-                            if (doCopy) {
+                            if (doCopy && !Files.isDirectory(file)) {
                                 Path target = classesDir.resolve(path);
                                 byte[] data = Files.readAllBytes(file);
```

Comment:
That's not part of your PR and I'm not even sure that's a problem since static files are usually small, but this won't work for large files. Is that an acceptable risk? There are two occurences of this method call in this class.

Reply:
Yes, the static file will have to be bigger than 2GB. I bet it is safe to keep it this way for now.

Reply:
If I'm not mistaken, the exact value depends on the heap size. But we're talking about dev mode and a very unlikely static file size, I agree to leave it that way.